vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/implementation/VLogReasoner.java

```diff
@@ -808,4 +813,59 @@ ReasonerState getReasonerState() {
 	void setReasonerState(ReasonerState reasonerState) {
 		this.reasonerState = reasonerState;
 	}
+
+	@Override
+	public Correctness writeInferences(OutputStream stream) throws IOException {
+		QueryResult queryAnswer;
+		Set<Predicate> toBeQueriedHeadPredicates = new HashSet<Predicate>();
+		TermQueryResultIterator stringQueryResultIterator;
+		QueryResultIterator answers;
+		for (Rule rule : this.knowledgeBase.getRules()) {
+			for (Literal literal : rule.getHead()) {
+				toBeQueriedHeadPredicates.add(literal.getPredicate());
+			}
+		}
+		for (DataSourceDeclaration dataSourceDeclaration : this.knowledgeBase.getDataSourceDeclarations()) {
+			toBeQueriedHeadPredicates.add(dataSourceDeclaration.getPredicate());
+		}
+		for (Fact fact : this.knowledgeBase.getFacts()) {
+			toBeQueriedHeadPredicates.add(fact.getPredicate());
+		}
+
+		for (Predicate predicate : toBeQueriedHeadPredicates) {
+			List<Term> toBeGroundedVariables = new ArrayList<Term>();
+			for (int i = 0; i < predicate.getArity(); i++) {
+				toBeGroundedVariables.add(Expressions.makeUniversalVariable("X" + i));
+			}
+			final karmaresearch.vlog.Atom vLogAtom = ModelToVLogConverter
+					.toVLogAtom(Expressions.makePositiveLiteral(predicate, toBeGroundedVariables));
+			try {
+				stringQueryResultIterator = this.vLog.query(vLogAtom, true, false);
+				answers = new VLogQueryResultIterator(stringQueryResultIterator, this.correctness);
+				while (answers.hasNext()) {
+					queryAnswer = answers.next();
+					try {
+						stream.write(Serializer.getFactString(predicate, queryAnswer.getTerms()).getBytes());
+					} catch (IOException e) {
+						throw new RuntimeException(e);
+					}
+				}
+			} catch (final NotStartedException e) {
+				throw new RuntimeException("Inconsistent reasoner state.", e);
+			} catch (final NonExistingPredicateException e1) {
+				LOGGER.warn("Query uses predicate " + predicate
+						+ " that does not occur in the knowledge base. Answer must be empty!");
+			}
+
+		}
+		return this.correctness;

```

Comment:
this method should behave similarily as answerQuery and exportQueryAnswersToCsv, and log a warning for incorrect answers.